### ISSUE TITLE
feat: Chat Widget Scroll improvement

### ISSUE THREAD

**Issue by sojan-official** [chatwoot/chatwoot#2330]
> During our testing of the chat widget we found that whenever new message comes, the widget scrolls to the bottom. But in some cases the message is huge in height and users often miss the questions and options at top. Will it be possible to scroll to the beginning of the first unseen message or a new message? It would improve UX a lot. 

The chatwidget should scroll to the first unseen message.

**Comment by anew-bhav** [comment]
@sojan-official I want to take this up. 
I have setup the environment successfully. Can you guide me through the next steps to solve this issue?

**Comment by sojan-official** [comment]
@pranavrajs  @nithindavid

**Comment by pranavrajs** [comment]
@anew-bhav 

The problem is that when a new message comes in a conversation in the dashboard, the scroll automatically happens to the bottom of the page skipping all the unread messages

In `components/widgets/conversation/MessagesView.vue` there is a method which `scrollToBottom` which scrolls all the way to the bottom of the page. It should actually scroll only to the `unread--toast` div. 

Please let me know if this makes sense.

**Comment by anew-bhav** [comment]
I was able to reproduce the issue by send a long text from the agent side to the widget.
I am new to vue, so things can be challenging. Anyhow, I complete the task. 
Please assign this issue to me.

**Comment by anew-bhav** [comment]
@pranavrajs 
With respect to what I have observed, I have created a schematic showing difference between the current and the required behavior. Let me know if that is correct. 

I am still struggling to understand, how should the `scrollDown` function be modified. 
The basic idea is, whenever `scrollDown` is called, the `conversation-panel` should scroll to the first unread message (if any), or the last messages top.

The challenge for me now is, how to work with different heights to make this happen. I need some assistance with this.

![2330-chat-widget-scroll-improvement](https://user-images.githubusercontent.com/18699797/121802885-69dbc480-cc5c-11eb-8937-a616518cf0a4.png)
